Mathlib/CategoryTheory/Sites/Sheafification.lean

```diff
@@ -166,14 +166,16 @@ def mk {X : C} {P : Cᵒᵖ ⥤ D} {S : J.Cover X} (x : Meq P S) : (J.plusObj P)
 theorem res_mk_eq_mk_pullback {Y X : C} {P : Cᵒᵖ ⥤ D} {S : J.Cover X} (x : Meq P S) (f : Y ⟶ X) :
     (J.plusObj P).map f.op (mk x) = mk (x.pullback f) := by
   dsimp [mk, plusObj]
-  simp only [← comp_apply, colimit.ι_pre, ι_colimMap_assoc]
+  erw [← comp_apply]
+  rw [ι_colimMap_assoc, colimit.ι_pre]
```

Comment:
Do you understand why you need `erw` instead of `simp`? This seems to be a common pattern in this PR.

Reply:
I can't seem to figure out why `erw` is needed here...

Reply:
My (unverified) guess is that `(Meq.equiv P S).symm x` is some element of `multiequalizer ...` whereas the morphism has a domain of the form `(diagram _ _).obj _`. The two are (obviously) defeq, but lean may have a hard time with the mismatch.

Reply:
So I guess there should be a (d)simp lemma that allows writing back and forth between the two exprs?

Reply:
Thanks to Adam's code, I found out that `rw [← comp_apply (x := (Meq.equiv P S).symm x) ...` works. So Lean needs to know about what `x` is